tests/test_terminal.py

```diff
@@ -258,3 +258,22 @@ async def test_culling(jp_server_config, jp_fetch):
             await asyncio.sleep(1)
 
     assert culled
+
+
+@pytest.mark.parametrize(
+    "terminado_settings,expected_shell",
+    [
+        ("shell_command=['/path/to/shell', '-l']", ["/path/to/shell", "-l"]),
```

Comment:
Using 1-element list here because once https://github.com/jupyter-server/jupyter_server_terminals/pull/42 gets merged and released I will add another test case of using a string in a follow-up.

Reply:
Released!

Reply:
Thanks, that was quick! Turns out parsing `Union`s from command line is broken in traitlets, I opened https://github.com/ipython/traitlets/pull/773 and tested that this test passes with traitlets patch.

Reply:
Just linking to a follow up to traitlets patch which is fixing some edge cases: https://github.com/ipython/traitlets/pull/774 for convenience.